### PR TITLE
Github actions: Add run unit test

### PR DESCRIPTION
**Pull request by ra-rau**
- Add a GitHub action to run unit tests.
- Closes issue#241

Signed-off-by: Ramona Rautenberg <rautenberg@osism.tech>

### CODE REVIEW

File: .github/workflows/run-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+name: Run Unit Tests
+
+"on":
+  push:
+    paths:
+      - src/**
+      - test/**
+      - .github/workflows/run-unit-tests.yml
```
Comment (berendt) [inline .github/workflows/run-unit-tests.yml:9]: Add requirements.txt and tox.ini

File: .github/workflows/run-unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    branches:
+      - main
+  pull_request:
+    paths:
+      - src/**
+      - test/**
+      - .github/workflows/run-unit-tests.yml
```
Comment (berendt) [inline .github/workflows/run-unit-tests.yml:16]: Add requirements.txt and tox.ini

**Review by berendt — CHANGES_REQUESTED** [review]

File: .github/workflows/run-unit-tests.yml (hunk 1 above)
Comment (ra-rau) [reply .github/workflows/run-unit-tests.yml:9]: done

File: .github/workflows/run-unit-tests.yml (hunk 2 above)
Comment (ra-rau) [reply .github/workflows/run-unit-tests.yml:18]: done

**Review by berendt — APPROVED** [review]

**Pull request merged**